packages/flutter/lib/src/painting/box_painter.dart

```diff
@@ -1421,12 +1421,14 @@ class _BoxDecorationPainter extends BoxPainter {
     );
   }
 
-  void _imageListener(ImageInfo value) {
+  void _imageListener(ImageInfo value, bool synchronousCall) {
     if (_image == value)
       return;
     _image = value;
-    assert(onChanged != null);
-    onChanged();
+    if (!synchronousCall) {
+      assert(onChanged != null);
```

Comment:
i'd put the assert outside the if statement. it's an invariant that should always be true.